draft-ietf-keytrans-protocol.md

```diff
@@ -1005,71 +1244,120 @@ that the result equals the root value of the prefix tree.
+the elements of the `prefix_proofs` field are search proofs from the prefix
+trees at particular log entries. There is no explicit indication as to which log
+entry the elements correspond to, as they are provided in the order that the
+algorithm the user is executing would request them. The elements of the
+`prefix_roots` field are, in left-to-right order, the prefix tree root
+hashes for any log entries whose timestamp was provided in `timestamps` but a
+search proof was not provided in `prefix_proofs`.
+
+If a log entry's timestamp is referenced multiple times, it is only added to
+`timestamps` the first time. Additionally, when a user advertises a
+previously-observed tree size in their request, log entry timestamps that the
+user is expected to have retained are excluded from `timestamps`.
+
+The `prefix_proofs` array differs from the `timestamps` array in two important
+ways. First, it is possible for the same log entry to have multiple entries in
+`prefix_proofs` if it is referenced multiple times. Users MUST verify that all
+`PrefixProof` structures corresponding to the same log entry compute the same
+prefix tree root hash. Second, while every log entry with a timestamp in
+`timestamps` will also be referenced in either `prefix_proofs` or
+`prefix_roots`, the inverse is not true. There may be elements of
+`prefix_proofs` or `prefix_roots` that correspond to log entries that are not in
+`timestamps` if the user is expected to have retained the log entry's timestamp.
+
+Users processing a `CombinedTreeProof` MUST verify that each field contains
+exactly the expected number of entries -- no more and no less.
+
+### Updating View
+
+For a user to update their view of the tree, the following is provided:
+
+- If the user has not previously observed a tree head, the timestamp of each log
+  entry along the frontier.
+- If the user has previously observed a tree head, the timestamps of each log
+  entry from the list computed in {{update-algorithm}}.
+
+Users verify that the timestamps represent a monotonic series, and that the
+rightmost timestamp is within the bounds defined by `max_ahead` and
+`max_behind`.
+
+### Fixed-Version Search
+
+For a user to search the combined tree for a specific version of a label, the
+following is provided:
+
+- For each log entry touched by the algorithm in {{fv-algorithm}}:
+  - The log entry's timestamp.
+  - If the log entry has surpassed its maximum lifetime and is on the frontier,
+    the right child's timestamp.
+  - If it is not the case that the log entry has surpassed its maximum lifetime,
+    is on the frontier, and the log entry's right child has also surpassed its
+    maximum lifetime, then a `PrefixProof` corresponding to a binary ladder in
+    the log entry's prefix tree is provided.
+- If the `PrefixProof` from the first log entry to contain the label-version
+  pair didn't touch the target version, provide a second `PrefixProof` from this
```

Comment:
What does "the label-version pair" refer to? The target? And what does "touch" mean? Doesn't every prefix proof "touch" the target as in it shows inclusion/non-inclusion?